kubeflow/optimizer/backends/kubernetes/backend.py

```diff
@@ -212,6 +213,60 @@ def get_job(self, name: str) -> OptimizationJob:
         optimization_job = self.__get_experiment_cr(name)
         return self.__get_optimization_job_from_cr(optimization_job)
 
+    def get_job_logs(
+        self,
+        name: str,
+        trial: Optional[str] = None,
+        follow: bool = False,
+        step: str = trainer_constants.NODE + "-0",
```

Comment:
Maybe for now, we should remove step from this API ?
The problem is that for other Steps (e.g. Pods), the container name is not `metrics-logger-and-collector`.
As a workaround, users can always use `TrainerClient()` to get logs for other steps of TrainJob
(e.g. TrainJob name == Trial name)